drivers/display/ek79652.c

```diff
@@ -0,0 +1,706 @@
+	if (err < 0) {
+		return err;
+	}
+	k_sleep(K_MSEC(EK79652_RESET_DELAY));
+	ek79652_busy_wait(dev);
+
+	LOG_DBG("Initialize EK79652 controller");
+
+	if (ek79652_write_cmd(dev, EK79652_CMD_PWR, config->pwr.data,
+				config->pwr.len)) {
+		return -EIO;
+	}
+
+	if (ek79652_write_cmd(dev, EK79652_CMD_BTST, config->softstart.data,
+					config->softstart.len)) {
+		return -EIO;
+	}
+
+	/* Power optimization */
+	if (config->pwropt.len) {
+		tmp[0] = config->pwropt.data[0];
+		tmp[1] = config->pwropt.data[1];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+
+		tmp[0] = config->pwropt.data[2];
+		tmp[1] = config->pwropt.data[3];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+
+		tmp[0] = config->pwropt.data[4];
+		tmp[1] = config->pwropt.data[5];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+
+		tmp[0] = config->pwropt.data[6];
+		tmp[1] = config->pwropt.data[7];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+
+		tmp[0] = config->pwropt.data[8];
+		tmp[1] = config->pwropt.data[9];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+
+		tmp[0] = config->pwropt.data[10];
+		tmp[1] = config->pwropt.data[11];
+		if (ek79652_write_cmd(dev, EK79652_CMD_PWROPT, tmp,
+			EK79652_PWROPT_LENGTH)) {
+			return -EIO;
+		}
+	}
+
+	/* Reset DFV_EN */
+	tmp[0] = EK79652_PDRF_VAL;
+	if (ek79652_write_cmd(dev, EK79652_CMD_PDRF, tmp, 1)) {
+		return -EIO;
+	}
+
+	/* Turn on: booster, controller, regulators, and sensor. */
+	if (ek79652_write_cmd(dev, EK79652_CMD_PON, NULL, 0)) {
+		return -EIO;
+	}
+	ek79652_busy_wait(dev);
+
+	/* Pannel settings : KW-BF or BWOTP-1F
+	 * KWR-AF BWROTP-0F
+	 */
+	if (config->lut_vcom_dc.len) {
+		tmp[0] = EK79652_PSR_RES1 | EK79652_PSR_LUT_EN | EK79652_PSR_BW |
+			EK79652_PSR_UD | EK79652_PSR_SHL |
+			EK79652_PSR_SHD | EK79652_PSR_RST;
+		if (ek79652_write_cmd(dev, EK79652_CMD_PSR, tmp, 1)) {
+			return -EIO;
+		}
+	} else {
+		tmp[0] = EK79652_PSR_BW |
+			EK79652_PSR_UD | EK79652_PSR_SHL |
+			EK79652_PSR_SHD | EK79652_PSR_RST;
+		if (ek79652_write_cmd(dev, EK79652_CMD_PSR, tmp, 1)) {
+			return -EIO;
+		}
+	}
+
+	tmp[0] = config->cdi;
+	if (ek79652_write_cmd(dev, EK79652_CMD_CDI, tmp, 1)) {
+		return -EIO;
+	}
+
+	if (config->lut_vcom_dc.len) {
+		set_partial_lut(dev);
+	}
+
+	if (ek79652_clear_display(dev, 0xFF, false)) {
+		return -1;
+	}
+
+	return 0;
+}
+
+static int ek79652_init(const struct device *dev)
+{
+	const struct ek79652_config *config = dev->config;
+	int err;
+
+	if (!spi_is_ready(&config->bus)) {
+		LOG_ERR("SPI bus %s not ready", config->bus.bus->name);
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(config->reset.port)) {
+		LOG_ERR("GPIO port for EK79652 reset is not ready");
+		return -EIO;
```

Comment:
Can you please use -ENODEV consistently when a device is not ready?

Reply:
will rectify in the following commit

Reply:
Stop to resolve unresolved conversations.